src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  return function add(num) {
+    if (num === undefined) {
+      const saveValue = result;
+
+      result = 0;
+
+      return saveValue;
+    }
+
+    result += num;
+
+    return add;
+  };
```

Comment:
use anonymous / arrow function
don't check for undefined - but provide an initial value instead
```suggestion
  return add = (num = 0) => {
    ...
  };
```